android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java

```diff
@@ -117,28 +116,9 @@ public void onDropViewInstance(ReactExoplayerView view) {
 
     @ReactProp(name = PROP_DRM)
     public void setDRM(final ReactExoplayerView videoView, @Nullable ReadableMap drm) {
-        if (drm != null && drm.hasKey(PROP_DRM_TYPE)) {
-            String drmType = ReactBridgeUtils.safeGetString(drm, PROP_DRM_TYPE);
-            String drmLicenseServer = ReactBridgeUtils.safeGetString(drm, PROP_DRM_LICENSE_SERVER);
-            ReadableArray drmHeadersArray = ReactBridgeUtils.safeGetArray(drm, PROP_DRM_HEADERS);
-            if (drmType != null && drmLicenseServer != null && Util.getDrmUuid(drmType) != null) {
-                UUID drmUUID = Util.getDrmUuid(drmType);
-                videoView.setDrmType(drmUUID);
-                videoView.setDrmLicenseUrl(drmLicenseServer);
-                if (drmHeadersArray != null) {
-                    ArrayList<String> drmKeyRequestPropertiesList = new ArrayList<>();
-                    for (int i = 0; i < drmHeadersArray.size(); i++) {
-                        ReadableMap current = drmHeadersArray.getMap(i);
-                        String key = current.hasKey("key") ? current.getString("key") : null;
-                        String value = current.hasKey("value") ? current.getString("value") : null;
-                        drmKeyRequestPropertiesList.add(key);
-                        drmKeyRequestPropertiesList.add(value);
-                    }
-                    videoView.setDrmLicenseHeader(drmKeyRequestPropertiesList.toArray(new String[0]));
-                }
-                videoView.setUseTextureView(false);
-            }
-        }
+        DRMProps drmProps = DRMProps.parse(drm);
+        videoView.setDrm(drmProps);
+        videoView.setUseTextureView(false);
```

Comment:
if drm props is null, we have to follow injected prop (useSecureView or useTextureView)

Reply:
It's hard question. because it depends on purpose of this library.
However, I prefer SurfaceView from a user perspective. In most situations, this is [beneficial](https://developer.android.com/media/media3/ui/playerview).
However, it can cause a lot of issues.

Decision is yours🫡

Reply:
I am working to remove this videoView.setUseTextureView(false);
It is a bit. The ugly solution is to add new variables recording the last calls to setUseTextureView and setUseSecureView and apply it in the else.

I would prefer implement it in another way:
- move DRM & viewType inside the source prop.
- > It will decrease the bridge usage, avoid restarting playback if we receive DRM prop after source and make cleaner code.

I will try to implement it, but I think we can merge this part now, and the next fix will come soon !

Reply:
@YangJonghun here is The patch I would to do for ensure we correctly reset the view state in the app:
https://github.com/TheWidlarzGroup/react-native-video/pull/3867
initially I didn't plan to it right now, but you make me accelerate :)
Still in draft as I need to do the same thing on ios, but that's the idea.
Let me know you thought please !

Reply:
I apologize for late response.
I left related comment in #3867 .